test/assertion_helper.rb

```diff
@@ -70,10 +69,10 @@ def inspect_source(source, cop, file = nil)
   end
 
   def investigate(cop, processed_source)
-    forces = RuboCop::Cop::Force.all.each_with_object([]) do |klass, instances|
-      next unless cop.join_force?(klass)
-
-      instances << klass.new([cop])
+    needed = Hash.new { |h, k| h[k] = [] }
+    Array(cop.class.joining_forces).each { |force| needed[force] << cop }
+    forces = needed.map do |force_class, joining_cops|
+      force_class.new(joining_cops)
```

Comment:
can you comment more on why you needed to make this change?

Reply:
Oh yes, sorry! I thought it was in a commit comment but I missed it.

Change was introduced in https://github.com/rubocop/rubocop/commit/f8813e76cef5c06cfe1f3a73b032f4f06de7de6c on v0.87. Details are vague, because it's a big commit/PR, but it removes the `join_force?` method in favor of `joining_forces`)

Upgrade notes:
https://github.com/rubocop/rubocop/blob/3482522e2be022eee2321722f4e049ab4bcb3bf1/docs/modules/ROOT/pages/v1_upgrade_notes.adoc#joining-forces

In reality, I just copy-pasted the updated version from https://github.com/rubocop/rubocop-minitest/blob/v0.11.0/test/assertion_helper.rb (As noted in the top of this file, the entire file is copied from there)